main.go

```diff
@@ -26,6 +28,24 @@ type Puzzle struct {
 	Solution []Placement
 }
 
+func (puz *Puzzle) String() string {
```

Comment:
this creates a compact string representation of a board, which can be used to unique represent a board when storing it in a database